ark-cli/src/commands/file/append.rs

```diff
@@ -35,12 +41,13 @@ impl Append {
             translate_storage(&Some(self.root_dir.to_owned()), &self.storage)
                 .ok_or(AppError::StorageNotFound(self.storage.to_owned()))?;
 
+        // FIXME: Why do we have `self.kind` and `self.storage`? Both are used to determine the storage type
```

Comment:
Kind is probably `file` or `folder`. It could be derived from names, at least for standard storages provided by ARK. But developers could also create their own storages which would be unknown to `ark-cli`, so the user would need to pass this information explicitly.

We could also figure out the storage type automatically by checking the path. But with atomic versions the path will always point to a folder, so we need to calculate depth of the folder. Folder storages would have depth of 3 (`storage/entry/version/value`), while file storage should have depth 2 (a file inside each version, i.e. `storage/version/table`).

Reply:
> Kind is probably `file` or `folder`.

Right.
- `storage`: Required parameter of type `String`. Gets passed to `translate_storage()` to determine the file path and `storage_type`
- `kind`: Optional parameter of type `StorageType` and specifies if it's a file or folder. It overwrites the value for `storage_type`

Why would we need `kind` if `storage` already determines the storage type? Are there any cases where using `kind` would come in handy?

Reply:
The only real use-case for `kind` is custom storages, i.e. when the storage is not shipped by ARK framework, but the user implemented their own instance of `FileStorage` or `FolderStorage`. In this case, we can't figure it out by the path. But we still can figure it out by the folder structure as I've described above.

Overall, `kind` flag is more like a temporary helper. We should determine it automatically.

Reply:
Got it.
I removed the FIXME since there is a reason to have both flags